engine/execution/computation/computer/computer.go

```diff
@@ -322,11 +322,11 @@ func (e *blockComputer) executeTransaction(
 	mergeSpan := e.tracer.StartSpanFromParent(txSpan, trace.EXEMergeTransactionView)
 	defer mergeSpan.Finish()
 
-	if tx.Err == nil {
-		err := collectionView.MergeView(txView)
-		if err != nil {
-			return err
-		}
+	// always merge the view, fvm take cares of reverting changes
+	// of failed transaction invocation
+	err = collectionView.MergeView(txView)
+	if err != nil {
+		return err
```

Comment:
May be worth wrapping error with more informative message

Reply:
added error wrapping at 6917ef9